examples/text-classification/run_glue.py

```diff
@@ -350,11 +350,29 @@ def compute_metrics(p: EvalPrediction):
 
     # Training
     if training_args.do_train:
-        trainer.train(
+        train_result = trainer.train(
             model_path=model_args.model_name_or_path if os.path.isdir(model_args.model_name_or_path) else None
         )
+        metrics = train_result.metrics
+        metrics["train_n_objs"] = len(train_dataset)
+
         trainer.save_model()  # Saves the tokenizer too for easy upload
 
+        output_train_file = os.path.join(training_args.output_dir, "train_results.txt")
+        if trainer.is_world_process_zero():
+            with open(output_train_file, "w") as writer:
+                logger.info("***** Train results *****")
+                for key, value in sorted(metrics.items()):
+                    logger.info(f"  {key} = {value}")
+                    writer.write(f"{key} = {value}\n")
+
+            # Need to save the state, since Trainer.save_model saves only the tokenizer with the model
+            trainer.state.save_to_json(os.path.join(training_args.output_dir, "trainer_state.json"))
+
+            # For convenience, we also re-save the tokenizer to the same directory,
+            # so that you can share your model easily on huggingface.co/models =)
+            tokenizer.save_pretrained(training_args.output_dir)
```

Comment:
No this is done by `trainer.save_model()`  at line 356 so there is no need to redo it.

Reply:
we are doing it in finetune_trainer - I'm just trying to sync. Let's pick one model and have others follow it.

I will remove it for now.